src/phasorpy/components.py

```diff
@@ -311,3 +315,209 @@ def graphical_component_analysis(
+        match the number of harmonics in the phasor coordinates.
+
+    See Also
+    --------
+    :ref:`sphx_glr_tutorials_api_phasorpy_components.py`
+
+    References
+    ----------
+    [2] Vallmitjana A, Lepanto P, Irigoin F, Malacrida L.
+    `Phasor-based multi-harmonic unmixing for in-vivohyperspectral imaging
+    <https://doi.org/10.1088/2050-6120/ac9ae9>`_.
+    *Methods Appl Fluoresc.*, (2022)
+
+    Example
+    -------
+    >>> n_fractions_from_phasor(
+    ...     [0.5, 0.3],
+    ...     [0.2, 0.7],
+    ...     components_real=[0.1, 0.3],
+    ...     components_imag=[0.2, 0.8],
+    ... )  # doctest: +NUMBER
+    (array([0.8162, 0.1513]), array([0.1958, 0.8497]))
+
+    """
+
+    real = numpy.atleast_1d(real)
+    imag = numpy.atleast_1d(imag)
+    components_real = numpy.atleast_1d(components_real)
+    components_imag = numpy.atleast_1d(components_imag)
+
+    if real.shape != imag.shape:
+        raise ValueError(f'{real.shape=} != {imag.shape=}')
+
+    if components_matrix is None:
+        if components_real is None or components_imag is None:
+            raise ValueError(
+                'either the components matrix or the components phasor '
+                'coordinates must be provided'
+            )
+        components_matrix = components_matrix_from_phasor(
+            components_real, components_imag
+        )
+
+    components_matrix = numpy.asarray(components_matrix)
+
+    num_harmonics = (components_matrix.shape[0] - 1) / 2
+    num_components = components_matrix.shape[1]
+
+    if components_matrix.shape[0] < num_components:
+        raise ValueError(
+            f'{components_matrix.shape[0]=} < {components_matrix.shape[1]=}'
+        )
+    if num_harmonics > 1 and num_harmonics != real.shape[0]:
+        raise ValueError(
+            f'number of harmonics in components matrix '
+            f'({num_harmonics}) does not match {real.shape[0]=}'
+        )
+
+    if real.ndim == 1:
+        real = numpy.expand_dims(real, axis=0)
+        imag = numpy.expand_dims(imag, axis=0)
+
+    invalid_mask = numpy.any(
+        numpy.isnan(real)
+        | numpy.isnan(imag)
+        | numpy.isinf(real)
+        | numpy.isinf(imag),
+        axis=0,
+    )
+    real = numpy.nan_to_num(real, nan=0.0, posinf=0.0, neginf=0.0)
+    imag = numpy.nan_to_num(imag, nan=0.0, posinf=0.0, neginf=0.0)
+
+    ones_array = numpy.ones((1,) + real.shape[1:])
+    coords = numpy.concatenate([real, imag, ones_array], axis=0)
+
+    fractions = scipy.linalg.lstsq(
+        components_matrix, coords.reshape(coords.shape[0], -1), **kwargs
+    )[0]
+    fractions = fractions.reshape((num_components,) + coords.shape[1:])
+    fractions = [numpy.where(invalid_mask, numpy.nan, f) for f in fractions]
+
+    return tuple(map(numpy.asarray, fractions))
+
+
+def components_matrix_from_phasor(
+    components_real: ArrayLike,
+    components_imag: ArrayLike,
+) -> NDArray[Any]:
```

Comment:
How about naming this function just `component_matrix` or `phasor_component_matrix`? Make argument positional:

```suggestion
def phasor_component_matrix(
    component_real: ArrayLike,
    component_imag: ArrayLike,
    /,
) -> NDArray[Any]:
```

Reply:
Thanks Christoph, this is a good suggestion. I will change this and push some other modifications I've been working on, also in the tutorial.

Reply:
I renamed to `components_matrix` to be consistent with the previous discussion, but if we decide to go with singular I can modify this also.